pages/api/program/index.ts

```diff
@@ -0,0 +1,64 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ResponseUtil } from "@utils/responseUtil";
+import { getPrograms, createProgram } from "@database/program";
+import { Program } from "models/Program";
+import { validateProgram } from "@utils/validation/program";
+
+/**
+ * handle controls the request made to the program resource
+ * @param req API request object
+ * @param res API response object
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    switch (req.method) {
+        case "GET": {
+            const programs = await getPrograms();
+            ResponseUtil.returnOK(res, programs);
+            res.status(200).json({});
+            break;
+        }
+        case "POST": {
+            // TODO:
+            if (!validateProgram(req.body as Program)) {
+                ResponseUtil.returnBadRequest(
+                    res,
+                    `Required field(s) were not provided correctly.`,
```

Comment:
If we could have validateProgram return a string/message regarding what was invalid, we could display that message here. That would be much more descriptive